source/NVDAObjects/UIA/excel.py

```diff
@@ -0,0 +1,458 @@
+		while parent.role == controlTypes.ROLE_TABLE:
+			parent = parent.parent
+		return parent._getUIACacheablePropertyValue(self._UIAExcelCustomProps.areGridLinesVisible.id)
+
+	#: Typing information for auto-property: _get_outlineColor
+	outlineColor: Optional[Tuple[colors.RGB]]
+
+	def _get_outlineColor(self) -> Optional[Tuple[colors.RGB]]:
+		val = self._getUIACacheablePropertyValue(_UIAHandler.UIA_OutlineColorPropertyId, True)
+		if isinstance(val, tuple):
+			return tuple(colors.RGB.fromCOLORREF(v) for v in val)
+		return None
+
+	#: Typing information for auto-property: _get_outlineThickness
+	outlineThickness: Optional[Tuple[float]]
+
+	def _get_outlineThickness(self) -> Optional[Tuple[float]]:
+		val = self._getUIACacheablePropertyValue(_UIAHandler.UIA_OutlineThicknessPropertyId, True)
+		if isinstance(val, tuple):
+			return val
+		return None
+
+	#: Typing information for auto-property: _get_fillColor
+	fillColor: Optional[colors.RGB]
+
+	def _get_fillColor(self) -> Optional[colors.RGB]:
+		val = self._getUIACacheablePropertyValue(_UIAHandler.UIA_FillColorPropertyId, True)
+		if isinstance(val, int):
+			return colors.RGB.fromCOLORREF(val)
+		return None
+
+	#: Typing information for auto-property: _get_fillType
+	fillType: Optional[_UIAConstants.FillType]
+
+	def _get_fillType(self) -> Optional[_UIAConstants.FillType]:
+		val = self._getUIACacheablePropertyValue(_UIAHandler.UIA_FillTypePropertyId, True)
+		if isinstance(val, int):
+			try:
+				return _UIAConstants.FillType(val)
+			except ValueError:
+				pass
+		return None
+
+	#: Typing information for auto-property: _get_rotation
+	rotation: Optional[float]
+
+	def _get_rotation(self) -> Optional[float]:
+		val = self._getUIACacheablePropertyValue(_UIAHandler.UIA_RotationPropertyId, True)
+		if isinstance(val, float):
+			return val
+		return None
+
+	#: Typing information for auto-property: _get_cellSize
+	cellSize: locationHelper.Point
+
+	def _get_cellSize(self) -> locationHelper.Point:
+		val = self._getUIACacheablePropertyValue(_UIAHandler.UIA_SizePropertyId, True)
+		x = val[0]
+		y = val[1]
+		return locationHelper.Point(x, y)
+
+	@script(
+		description=pgettext(
+			"excel-UIA",
+			# Translators: the description of a script
+			"Shows a browseable message Listing information about a cell's "
+			"appearance such as outline and fill colors, rotation and size"
+		),
+		gestures=["kb:NVDA+o"],
+	)
+	def script_showCellAppearanceInfo(self, gesture):
+		infoList = []
+		tmpl = pgettext(
+			"excel-UIA",
+			# Translators: The width of the cell in points
+			"Cell width: {0.x:.1f} pt"
+		)
+		infoList.append(tmpl.format(self.cellSize))
+
+		tmpl = pgettext(
+			"excel-UIA",
+			# Translators: The height of the cell in points
+			"Cell height: {0.y:.1f} pt"
+		)
+		infoList.append(tmpl.format(self.cellSize))
+
+		if self.rotation is not None:
+			tmpl = pgettext(
+				"excel-UIA",
+				# Translators: The rotation in degrees of an Excel cell
+				"Rotation: {0} degrees"
```

Comment:
Have you considered npgettext here, degree vs degrees?

Reply:
I don't believe we have any other code in NVDA that uses this? Do we know this works okay with poedit / the rest of our translation system?
Perhaps something to test outside of this pr.

Reply:
I tested it with #11598 as per @feerrenrut's request after @bramd suggested it

Reply:
Ah good to know. However, in this case rotation is a float not an int, so even for 1 degree it will be 1.0 degrees.

Reply:
I'm convinced, feel free to leave it as is 😊